mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala

```diff
@@ -176,27 +177,47 @@ class MinMaxScalerModel private[ml] (
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     transformSchema(dataset.schema, logging = true)
-    val originalRange = (originalMax.asBreeze - originalMin.asBreeze).toArray
+
+    val numFeatures = originalMax.size
+    val scale = $(max) - $(min)
+    val minValue = $(min)
+
+    // transformed value for constant cols
+    val constantOutput = 0.5 * scale + minValue
```

Comment:
I suppose this could also be `($(max) + $(min)) / 2.0`, might be clearer, but it's the same mathematically.